File: CarlaErik/UvA.SoftCon.Questionnaire/AST/Model/Expressions/BinaryExpression.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using UvA.SoftCon.Questionnaire.Utilities;

namespace UvA.SoftCon.Questionnaire.AST.Model.Expressions
{
    public class BinaryExpression : Node, IExpression
    {
        public override NodeType Type
        {
            get
            {
                return NodeType.BinaryExpression;
            }
        }

        public Operation Operation
        {
            get; 
            private set;
        }

        public IExpression Left
        {
            get;
            private set;
        }

        public IExpression Right
        {
            get;
            private set;
        }

        public BinaryExpression(Operation operation, IExpression left, IExpression right, TextPosition position)
            : base(position)
        {
            Operation = operation;
            Left = left;
            Right = right;
        }

        public override void Accept(IASTVisitor visitor)
        {
            visitor.Visit(this);
        }

        public override string ToString()
        {
            return String.Format("{0} {1} {2}", Left.ToString(), StringEnum.GetStringValue(Operation), Right.ToString());
        }
    }
}

File: Nian_Karl/QL_Nian_Karl/src/org/uva/sea/ql/AST/expression/BinaryExpression.java
package org.uva.sea.ql.AST.expression;

import org.uva.sea.ql.AST.visitor.Visitor;


public abstract class BinaryExpression extends Expression{
	protected final Expression leftExpression;
	protected final Expression rightExpression;
	
	public BinaryExpression(Expression leftExpression,
			Expression rightExpression) {
		this.leftExpression = leftExpression;
		this.rightExpression = rightExpression;
	}
	
	public Expression getLeftExpression() {
		return leftExpression;
	}


	public Expression getRightExpression() {
		return rightExpression;
	}
	@Override
	public abstract void accept(Visitor visitor);
	
}
